src/core/source_buffers/segment_inventory.ts

```diff
@@ -655,3 +619,166 @@ export default class SegmentInventory {
     return this.inventory;
   }
 }
+
+/**
+ * Returns `true` if the buffered start of the given chunk looks coherent enough
+ * relatively to what is anounced in the Manifest.
+ * @param {Object} thisSegment
+ * @returns {Boolean}
+ */
+function bufferedStartLooksCoherent(
+  thisSegment : IBufferedChunk
+) : boolean {
+  if (thisSegment.bufferedStart === undefined ||
+      !thisSegment.isCompleteSegment)
+  {
+    return false;
+  }
+  const { start, end } = thisSegment;
+  return Math.abs(start - thisSegment.bufferedStart) <=
+           MAX_MANIFEST_BUFFERED_START_END_DIFFERENCE &&
+         (thisSegment.bufferedEnd === undefined ||
+           thisSegment.bufferedEnd > thisSegment.bufferedStart &&
+           Math.abs(thisSegment.bufferedEnd - thisSegment.bufferedStart -
+             (end - start)) <= MAX_MANIFEST_BUFFERED_DURATION_DIFFERENCE);
+
+         // thisSegment.bufferedEnd != null &&
+         // thisSegment.bufferedEnd > thisSegment.bufferedStart &&
+         // Math.abs(thisSegment.bufferedEnd - thisSegment.bufferedStart -
+         //   (end - start)) <= MAX_MANIFEST_BUFFERED_DURATION_DIFFERENCE;
+}
+
+/**
+ * Returns `true` if the buffered end of the given chunk looks coherent enough
+ * relatively to what is anounced in the Manifest.
+ * @param {Object} thisSegment
+ * @returns {Boolean}
+ */
+function bufferedEndLooksCoherent(
+  thisSegment : IBufferedChunk
+) : boolean {
+  if (thisSegment.bufferedEnd === undefined ||
+      !thisSegment.isCompleteSegment)
+  {
+    return false;
+  }
+  const { start, end } = thisSegment;
+  return Math.abs(end - thisSegment.bufferedEnd) <=
+           MAX_MANIFEST_BUFFERED_START_END_DIFFERENCE &&
+         thisSegment.bufferedStart != null &&
+         thisSegment.bufferedEnd > thisSegment.bufferedStart &&
+         Math.abs(thisSegment.bufferedEnd - thisSegment.bufferedStart -
+           (end - start)) <= MAX_MANIFEST_BUFFERED_DURATION_DIFFERENCE;
```

Comment:
So, the `MAX_MANIFEST_BUFFERED_DURATION_DIFFERENCE` is not relative ?
It can represent a huge proportion of loss for small segments.